### DISCUSSION TITLE
Working around stable-looking nightly versions in Maven

### DISCUSSION THREAD

**Discussion by mattTea** [renovatebot/renovate#20771] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

We have just started using Renovate and pretty much all our tests look grand (great job Renovate maintainers!)

One library seems to be picking up a snapshot-type (unstable) version though

The library is [graphql-java-extended-scalars](https://github.com/graphql-java/graphql-java-extended-scalars)
- Our starting version was 19.2
- Current latest stable version (expected upgrade) is 20.0
- Renovate is picking up and upgrading to 2023-01-24T02-11-56-babda5f
- This version I have since found out is a master build version, not the stable version, as per this [discussion](https://github.com/graphql-java/graphql-java/discussions/3106)

### What you tried so far.

I have created a [minimal reproduction](https://gitlab.com/renovate-version-issue)
- You can see the merge for the update [here](https://gitlab.com/renovate-version-issue/renovate-graphql-version-issue/-/commit/6b641c62d1384e0f41edebcff92270a591e92a46)

To replicate...
- Set the version in the build.gradle.kts file back to 19.1
- Run the [scheduled pipeline](https://gitlab.com/renovate-version-issue/renovate/-/pipeline_schedules) by clicking the play button, and watch the merge requests in [this project](https://gitlab.com/renovate-version-issue/renovate-graphql-version-issue/-/merge_requests?scope=all&state=all)

**Comment by rarkins** [comment, marked as answer]
You can add a `packageRules` entry with `matchPackageNames=com.graphql-java:graphql-java-extended-scalars` and `allowedVersions=<2000`.

**Comment by rarkins** [comment]
Idea for solving in future: #20772

**Comment by mattTea** [comment]
Thanks @rarkins, we're doing something similar

Our small team have over 50 repos running Renovate on a daily schedule and would rather not have to check each update to ensure it's picked up a stable release, so were hoping Renovate would pick up the stable versions by default

(Just seen your idea above though, looks like a good option 👍)

**Reply by rarkins** [reply]
@mattTea important to know because you are new to Renovate: Renovate always tries to pick up only stable versions by default. Versions like `2023-01-24T02-11-56-babda5f` are stable according to the Maven spec because it's very flexible, and this label is using none of the syntax to indicate an unstable release. 

Renovate is doing nothing wrong, and the feature is to work around other's mistakes/problems.

I'm going to correct the title of this discussion so that it does not confuse any future users
